dango/dex/src/execute.rs

```diff
@@ -348,6 +357,74 @@ fn withdraw_liquidity(
     // TODO: add events
 }
 
+#[inline]
+fn swap_exact_amount_in(
+    ctx: MutableCtx,
+    route: UniqueVec<PairId>,
+    minimum_output: Option<Uint128>,
+) -> anyhow::Result<Response> {
+    let (reserves, output) =
+        core::swap_exact_amount_in(ctx.storage, route, ctx.funds.into_one_coin()?)?;
+
+    // Ensure the output is above the minimum.
+    if let Some(minimum_output) = minimum_output {
+        ensure!(
+            output.amount >= minimum_output,
+            "output amount is below the minimum: {} < {}",
+            output.amount,
+            minimum_output
+        );
+    }
+
+    // Save the updated pool reserves.
+    for (pair, reserve) in reserves {
+        RESERVES.save(ctx.storage, (&pair.base_denom, &pair.quote_denom), &reserve)?;
+    }
+
+    // TODO:
+    // 1. add events
+    // 2. handle the case if output is zero
```

Comment:
should we just set `minimum_out = minimum_out.unwrap_or(Uint128::ZERO)?`? then the tx will fail if output is zero. seems most reasonable to me

Reply:
@larry0x I implemented it like this now